pkg/flow/internal/controller/loader.go

```diff
@@ -202,23 +209,23 @@ func (l *Loader) EvaluateDependencies(parentContext *hcl.EvalContext, c *Compone
 			// arguments will need re-evaluation.
 			return nil
 		}
-		l.evaluate(parentContext, n.(*ComponentNode), true, false)
-		return nil
+		return l.evaluate(parentContext, n.(*ComponentNode), true, false)
```

Comment:
We need to put some thought into whether this is correct or not. Anything returning an error during Walk will cause the Walk to abort. This means that if an evaluation fails, any dependency won't be re-evaluated, even if some of its expressions would've properly changed values. 

We could just ignore the evaluation here too; at this point we're probably more interested in whether errors cause a component to go unhealthy than acting on a failed eval.